modules/dnn/src/onnx/onnx_graph_simplifier.cpp

```diff
@@ -145,13 +145,153 @@ class SoftMaxSubgraph : public Subgraph
 };
 
+class UpsampleSubgraph : public Subgraph
+{
+public:
+    UpsampleSubgraph()
+    {
+        int input = addNodeToMatch("");
+
+        int indices1 = addNodeToMatch("Constant");
+        int inpShape1 = addNodeToMatch("Shape", input);
+        int inpGather1 = addNodeToMatch("Gather", inpShape1, indices1);
+        int inpCastG1 = addNodeToMatch("Cast", inpGather1);
+        int B1 = addNodeToMatch("Constant");
+        int inpMul1 = addNodeToMatch("Mul", inpCastG1, B1);
+        int inpCastM1 = addNodeToMatch("Cast", inpMul1);
+        int inpFloor1 = addNodeToMatch("Floor", inpCastM1);
+
+        int indices2 = addNodeToMatch("Constant");
+        int inpShape2 = addNodeToMatch("Shape", input);
+        int inpGather2 = addNodeToMatch("Gather", inpShape2, indices2);
+        int inpCastG2 = addNodeToMatch("Cast", inpGather2);
+        int B2 = addNodeToMatch("Constant");
+        int inpMul2 = addNodeToMatch("Mul", inpCastG2, B2);
+        int inpCastM2 = addNodeToMatch("Cast", inpMul2);
+        int inpFloor2 = addNodeToMatch("Floor", inpCastM2);
+
+        int inpUnsqueeze1 = addNodeToMatch("Unsqueeze", inpFloor1);
+        int inpUnsqueeze2 = addNodeToMatch("Unsqueeze", inpFloor2);
+        int inpConcat1 = addNodeToMatch("Concat", inpUnsqueeze1, inpUnsqueeze2);
+
+        int inpIndex = addNodeToMatch("Constant");
+
+        int inpCast3 = addNodeToMatch("Cast", inpConcat1);
+
+        int inpShape3 = addNodeToMatch("Shape", input);
+        int inpSlice = addNodeToMatch("Slice", inpShape3);
+        int inpCast4 = addNodeToMatch("Cast", inpSlice);
+
+        int inpDiv = addNodeToMatch("Div", inpCast3, inpCast4);
+        int inpConcat2 = addNodeToMatch("Concat", inpIndex, inpDiv);
+
+        addNodeToMatch("Upsample", input, inpConcat2);
+        setFusedNode("Upsample", input, indices1, B1, indices2, B2, inpIndex);
+    }
+
+    virtual void finalize(const Ptr<ImportGraphWrapper>&,
+                          const Ptr<ImportNodeWrapper>& fusedNode,
+                          std::vector<Ptr<ImportNodeWrapper> >& inputs) CV_OVERRIDE
+    {
+        opencv_onnx::NodeProto* node = fusedNode.dynamicCast<ONNXNodeWrapper>()->node;
+        CV_Assert(node->input_size() == 6);
+
+        std::vector<opencv_onnx::NodeProto*> input_nodes(3);
+        std::vector<cv::Mat> nodes_attributes(3);
+        input_nodes[0] = inputs[5].dynamicCast<ONNXNodeWrapper>()->node; //inpIndex
+        input_nodes[1] = inputs[2].dynamicCast<ONNXNodeWrapper>()->node; //B1
+        input_nodes[2] = inputs[4].dynamicCast<ONNXNodeWrapper>()->node; //B2
+        for(int i = 0; i < 3; i++)
+        {
+            opencv_onnx::TensorProto tensor_proto = input_nodes[i]->attribute(0).t();
+            nodes_attributes[i] = getMatFromTensor(tensor_proto);
+        }
+        CV_Assert((int)nodes_attributes[0].at<float>(0) == 1 && (int)nodes_attributes[0].at<float>(1) == 1 &&
+            nodes_attributes[1].total() == 1 && nodes_attributes[2].total() == 1);
```

Comment:
I'd like to recommend to simplify it.

`nodes_attributes[0]` is never used and I believe that we can extract values from B1 and B2 without `getMatFromTensor` (maybe not, depends of field type). Some thing like `B1->float(0)` or similar.

Reply:
```
if (datatype == opencv_onnx::TensorProto_DataType_FLOAT) {

        if (!tensor_proto.float_data().empty()) {
            const ::google::protobuf::RepeatedField<float> field = tensor_proto.float_data();
            Mat(sizes, CV_32FC1, (void*)field.data()).copyTo(blob);
        }
        else {
            char* val = const_cast<char*>(tensor_proto.raw_data().c_str());
            Mat(sizes, CV_32FC1, val).copyTo(blob);
        }
    }
```
B1 falls under second category, i.e `raw_data()` so I couldn't find any simple function to extract it. Please recommend I missed it.

Reply:
`nodes_attribute[0]` has scales of batch and channels. [batch, channels, Height, width]. I understand that it is always [1,1, scale_height, scale_width]. But I still kept it to check in case they are not 1. 
Not sure to keep them or remove them. What do you say?

Reply:
Ok, then let's rename them properly:

```
setFusedNode("Upsample", input, ?, scaleH, ?, scaleW, scaleNC);
```
BTW, what is a size of `nodes_attributes[0]`?

Reply:
@dkurt, can I solve two issues in this PR. issue [16475](https://github.com/opencv/opencv/issues/16475) has similar subgraph of Resize.
![Screenshot from 2020-02-16 18-57-25](https://user-images.githubusercontent.com/26292819/74605464-5d8a3a80-50ee-11ea-83a8-d6557489cfea.png)
while current subgraph of upsample
![Screenshot from 2020-02-13 17-43-16](https://user-images.githubusercontent.com/26292819/74605480-7a267280-50ee-11ea-87b5-b3afebfe7d00.png)
and we may find some other variations in future. So, I am thinking of creating a layer of this part of subgraph 
![Screenshot from 2020-02-16 19-00-03](https://user-images.githubusercontent.com/26292819/74605544-ffaa2280-50ee-11ea-94b9-4eef026c377a.png)

which is common in both and will use this fused as a part of both subgraphs.

Reply:
> Ok, then let's rename them properly:
> 
> ```
> setFusedNode("Upsample", input, ?, scaleH, ?, scaleW, scaleNC);
> ```
> 
> BTW, what is a size of `nodes_attributes[0]`?

`nodes_attributes[0]` has [1.0, 1.0].

Reply:
Both subgraphs are obtained from same pytorch layer `Upsample` but with different versions. version of subgraph of current issue is 9 whereas vesion of subgraph of issue [16475](https://github.com/opencv/opencv/issues/16475) is 11.

Reply:
@ashishkrshrivastava, if you don't mind, let's enable these two subgraph in this PR. This way it could help to see how they similar.

Reply:
@dkurt , I am not able to produce subgraph same as in model in [16475](https://github.com/opencv/opencv/issues/16475). I am getting this subgraph with the same code which is little different.
![Screenshot from 2020-02-16 22-22-31](https://user-images.githubusercontent.com/26292819/74608886-e911c480-510a-11ea-82e0-15719988ca1d.png)

I have asked the way that model has exported in comments there. Will work on that issue later. 
Actually, slight variation in pytorch version or onnx version are producing different subgraphs. Even sometimes sequential and functional way of pytorch gives different subgraph. 
Please continue with this one only.

Reply:
Ok, got it, thanks!

So let's finish this PR first just not to keep your waiting